### ISSUE TITLE
Follow hinting doesn't work on miniflux

### ISSUE THREAD

**Issue by sprnza** [luakit/luakit#708]
Hi there!
I've set up my own miniflux instance and have found out that luakit is not able to hint URLs on a page that miniflux provides. Here is the link to my [miniflux](http://news.speran.info) Login: luakit, Password: luakit. Feel free to use it for testing.

**Comment by amosbird** [comment]
Hi, iirc, that's the limitation of current hinting engine in luakit (manipulating DOM). There isn't much choice left unless a complete rewrite.

**Comment by aidanholm** [comment]
Yep, @amosbird is correct; specifically, the CSS that styles the hint elements is getting rejected due to minflux's content security policy, since this CSS is applied via an inline element.

A possible workaround is to remove custom follow mode CSS, and just keep it always applied as a userstyle. Unfortunately this would break the follow mode API, but I don't think it's widely used (or used at all).

**Comment by sprnza** [comment]
Hi @aidanholm! Thanks for the comment! Is the workaround for devs or is it something could be applied locally?

**Comment by aidanholm** [comment]
By 'workaround' I actually mean 'bugfix', so it'd need some dev work. Unfortunately CSP also applies to inline styles, so probably the HTML overlay design should just be scrapped entirely (and reimplemented).

**Comment by dlobue** [comment]
@aidanholm I found another website where follow hinting doesn't work, the python packaging search page: https://pypi.org/search/?q=docopt
Is follow hinting not working there for the same reason as miniflux?

**Comment by jun7** [comment]
Sorry. I never noticed this feature of github.
The commit above does not relate luakit.

**Comment by aidanholm** [comment]
@dlobue yep, same reason.
@jun7 no worries; wyeb looks like an interesting project!

the approach used in that commit is similar to what luakit would have to do to fix this issue, but the plugin-based approach of luakit will complicate things somewhat

**Comment by jun7** [comment]
I had thought luakit wills choose another way but
so, I'm happy it was not a complete misplacement.
Thank you!

**Comment by hollmmax** [comment]
[Archlinux packages](https://www.archlinux.org/packages) are also on the list...
Do we have a list of broken pages, there seems to be a lot of them.

**Comment by Jab2870** [comment]
Just wanted to see if there had been any update on this? It's becoming more and more of a problem as sites increasingly adopt strong CSPs.

**Comment by balejk** [comment]
Hi, could you please see if #1125 fixes the issue for you?

**Comment by fictitiousexistence** [comment]
https://github.com/luakit/luakit/pull/1125 fixes the issue in miniflux for me
